src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -24,6 +24,7 @@
  */
 
 #include "precompiled.hpp"
+#include "jfr/periodic/sampling/jfrCPUTimeThreadSampler.hpp"
```

Comment:
We try to channel all (most) VM code through the jfr.hpp as a single entry point to the JFR system. Can these hooks be incorporated in the JfrThreadLocal::on_start() and JfrThreadLocal::on_exit()?

If so, you don't need to touch anything in javaThread.cpp.